x-pack/plugins/ml/public/application/components/color_range_legend/use_color_range.ts

```diff
@@ -186,3 +186,7 @@ export const useColorRange = (
 
   return scaleTypes[colorRangeScale];
 };
+
+export function useCurrentEuiTheme() {
+  return { euiTheme: useUiSettings().get('theme:darkMode') ? euiThemeDark : euiThemeLight };
```

Comment:
Bear in mind it returns a new instance on each render. You probably should wrap in with `useMemo`.

Reply:
using `useMemo` threw errors about not having hooks inside `useMemo`

Reply:
have you tried this? 
```suggestion
  const uiSettings = useUiSettings();
  return useMemo(() => ({ euiTheme: uiSettings.get('theme:darkMode') ? euiThemeDark : euiThemeLight }), [uiSettings])

```